api/core/v1alpha2/virtual_machine_snapshot.go

```diff
@@ -0,0 +1,131 @@
+// +groupName=virtualization.deckhouse.io
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	VirtualMachineSnapshotKind     = "VirtualMachineSnapshot"
+	VirtualMachineSnapshotResource = "virtualmachinesnapshots"
+)
+
+// VirtualMachineSnapshot provides a resource for creating snapshots of virtual machines.
+//
+// +kubebuilder:object:root=true
+// +kubebuilder:metadata:labels={heritage=deckhouse,module=virtualization}
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=virtualization,scope=Namespaced,shortName={vmsnapshot,vmsnapshots},singular=virtualmachinesnapshot
+// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="VirtualMachineSnapshot phase."
+// +kubebuilder:printcolumn:name="Consistent",type="boolean",JSONPath=".status.consistent",description="VirtualMachineSnapshot consistency."
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="VirtualMachineSnapshot age."
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineSnapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VirtualMachineSnapshotSpec   `json:"spec"`
+	Status VirtualMachineSnapshotStatus `json:"status,omitempty"`
+}
+
+// VirtualMachineSnapshotList contains a list of `VirtualMachineSnapshot`
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []VirtualMachineSnapshot `json:"items"`
+}
+
+type VirtualMachineSnapshotSpec struct {
+	// The name of virtual machine to take snapshot.
+	//
+	// +required
+	VirtualMachineName string `json:"virtualMachineName"`
+	// Create a snapshot of a virtual machine only if it is possible to freeze the machine through the agent.
+	//
+	// If value is true, the snapshot of the virtual machine will be taken only in the following scenarios:
+	// - the virtual machine is powered off.
+	// - the virtual machine with an agent, and the freeze operation was successful.
+	//
+	// +kubebuilder:default:=true
+	RequiredConsistency bool `json:"requiredConsistency"`
+	// +kubebuilder:default:="Always"
+	KeepIPAddress KeepIPAddress `json:"keepIPAddress"`
+	// +optional
+	VolumeSnapshotClassNames []VolumeSnapshotClassName `json:"volumeSnapshotClassNames"`
+}
+
+type VirtualMachineSnapshotStatus struct {
+	Phase VirtualMachineSnapshotPhase `json:"phase"`
+	// The virtual machine snapshot is consistent.
+	Consistent *bool `json:"consistent,omitempty"`
+	// The name of underlying `Secret`, created for virtual machine snapshotting.
+	VirtualMachineSnapshotSecretName string `json:"virtualMachineSnapshotSecretName,omitempty"`
+	// The list of `VirtualDiskSnapshot` names that associated with the virtual machine and taken for snapshotting.
```

Comment:
```suggestion
	// The list of `VirtualDiskSnapshot` names associated with the virtual machine disks to snapshot them.
```
Is it better?

Reply:
The list of `VirtualDiskSnapshot` names for the snapshots taken from the virtual disks of the associated virtual machine."